src/compiler/scala/tools/nsc/backend/jvm/BCodeHelpers.scala

```diff
@@ -35,15 +35,26 @@ abstract class BCodeHelpers extends BCodeIdiomatic with BytecodeWriters {
   /*
    * must-single-thread
    */
-  def getOutFolder(csym: Symbol, cName: String, cunit: CompilationUnit): _root_.scala.tools.nsc.io.AbstractFile = {
-    try {
-      outputDirectory(csym)
-    } catch {
-      case ex: Throwable =>
-        reporter.error(cunit.body.pos, s"Couldn't create file for class $cName\n${ex.getMessage}")
-        null
-    }
-  }
+  def getOutFolder(csym: Symbol, cName: String, cunit: CompilationUnit): _root_.scala.tools.nsc.io.AbstractFile = Option {
+      try {
+        outputDirectory(csym)
+      } catch {
+        case ex: Throwable =>
+          reporter.warning(cunit.body.pos, s"Couldn't find output folder for symbol source ${csym.name}. Dropping to compliation unit source.\n${ex.getMessage}")
```

Comment:
About first clause, you are right. There warnings will break the build if fatal warnings is on. I'm not sure what level is ok. Done a little bit defensive as delambdafy and GenBCode are `Y`. But maybe I'm wrong. So let me know :).
The second clause... I'm not too good in area of symbols. I tried to fix the missing assiciated file in Delamdbafy.scala file as I described it in PR description above. The problem is that `_baz` is, as I remember, a TermSymbol and is treated as Class Def for lambda and it doesn't have associatedFile field I guess. I try to think it over again.